src/storage/gcloud-storage/gcp-storage.service.ts

```diff
@@ -119,6 +121,41 @@ export class GCPStorageService {
         return ok(true);
     }
 
+    private async compressFile(file: File): Promise<Buffer> {
+        const chunks: Buffer[] = [];
+
+        for await (const chunk of file.stream()) {
+            chunks.push(Buffer.from(chunk));
+        }
+        const fileBuffer = Buffer.concat(chunks);
+
+        try {
+            const sharpPipeline = sharp(fileBuffer).resize({
+                width: 1200,
+                withoutEnlargement: true,
+            });
+
+            const format = file.type.split('/')[1];
```

Comment:
Should we `toLowerCase` this incase an upload is named like `foo.JPEG`?

Reply:
or alternatively doesn't sharp provide some way of detecting the format of the image from the contents rather than the name?

Reply:
^ this is using the file MIME type, so e.g. `image/jpeg` for a JPG (not the file extension, e.g. `my-image.jpg`)

There is a validation on the file MIME types in the `validateFile` method ([code](https://github.com/TryGhost/ActivityPub/blob/2465f741cd52340431768bdf748aafef4159277d/src/storage/gcloud-storage/gcp-storage.service.ts#L110)) that allows only the following:
```
const ALLOWED_IMAGE_TYPES = [
    'image/jpg',
    'image/jpeg',
    'image/png',
    'image/webp',
];
```

Reply:
Ah my bad, all good!